### PR TITLE
Closes #1105 #1262 restructure docs

### LINKED ISSUE #1105: Documentation:  Restructure Reference Page
### Please select a category the issue is focused on?

Other

### Let us know where something needs a refresh or put your idea here!

The reference page should be restructured:
- Remove “Other Keywords”, move “Timing”, “Sources” to User Utilities, move TTE sources and Metadata to top level
- Add subsection for User Utilities and Developer Utilities/ Various, User Utilities should contain functions called in ADaM scripts, Developer utilities should contain functions called in admiral functions or functions of admiral extension packages. User Utilities should not contain `derive_*` functions.
- Remove "Miscellaneous" section

### PR DESCRIPTION
**Pull request by bms63**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
Hi all,

I've moved this along as much as I can.  Please see the published website with my updates to all the R functions, Reference page and to the Programming Strategy.  I think there is still some updates to made with our vignettes

[bms63.github.io/admiral/](https://bms63.github.io/admiral/)

I've created an excel file that has all the R functions pinned to their R file locations and what I think their keywords and family should be.  IF you think something is not going to the right place, please flag that for me.  We can change this as we go as we have two months before release so we can work on the kinks!

We can make new families and keywords, but I'm really leaning towards everything being mutually exclusive to reduce the size of the Reference page and make it easy on developers to just pick one keyword/family.  

https://exbpbox.box.com/s/aj6t1c2tokoxs0g2a7in1yu7ioy136gk
image

**Comment by bms63** [comment]
Oh!  And i was hoping folks could review before tomorrows general meeting :)

**Comment by bundfussr** [comment]
I am not sure if we should implement it this way. In my opinion it does not make a big difference regarding findability. The derivations and computations were restructured but I think this was not the problematic part. For the derivations and computations both structures the old and the new one are fine. So I wonder if we should change them if there is no big benefit.

The main goal of the issue was to get rid of not meaningful sections like "Miscellaneous" and structure too general sections like "User Utilities". But we still have these in the new structure.

**Comment by rossfarrugia** [comment]
i feel like it is a step in the right direction, but also i don't think we should be the ones to decide as we already know all these functions so our perspectives are already biased. we just ctrl+f to get the ones we're looking for.

@thomas-neitmann why not ask some of our early adopter teams to take a look and compare old v new and let us know their thoughts? i'd rather follow the users here, especially newer ones still trying to navigate the package. i'm sure there will be lots of differences of opinions as well, and that's where we ultimately can make a call, as for things like this you'll never find a solution that keeps all happy.

**Comment by bms63** [comment]
> I am not sure if we should implement it this way. In my opinion it does not make a big difference regarding findability. The derivations and computations were restructured but I think this was not the problematic part. For the derivations and computations both structures the old and the new one are fine. So I wonder if we should change them if there is no big benefit.
> 

So I would go for new (as I am biased :)) and keep everything mutually exclusive.  Currently the keywords are all over the place.  My update reduces the scatterness of the keywords and aligns them 1-1 to the families.  This makes it easier for users to find similar functions in the documentation and lessens the burdens on developers to make choices on what keywords should be used, i.e. they just pick one keyword/family group and are done with it.

image

> The main goal of the issue was to get rid of not meaningful sections like "Miscellaneous" and structure too general sections like "User Utilities". But we still have these in the new structure.


I've removed the Miscellaneous structure in the updated version.  - https://bms63.github.io/admiral/reference/index.html

The User Utilities can be broken down into four sections:  Utilities for Dataset Checking,  Utilities for Filtering Observations, Utilities for Formatting Observations and Helper Utilities.  @rossfarrugia had asked for this to be removed, but I can put back in.  I realize Helper Utilities could be re-defined.


I'm thinking that a new developer utility should be created that can extract all our functions and make sure they are classified correctly.   I've made this for myself, but as we add and remove things keeping an updated spreadsheet of all our functions is really important to help with classification and keeping our documentation in tip top shape.

**Comment by bms63** [comment]
@pharmaverse/admiral Hi all,  I'd like to resolve this issue of restructuring the documentation.  Hoping it can be the first item on our agenda tomorrow.

**Comment by bms63** [comment]
It's me!  I've made some updates for the Reference page. 

https://bms63.github.io/admiral/reference/index.html
- Re-split the Utilities into different subgroups as requested.
- Created a new Utilities group for Examples
- Rename family from `occds` to `der_occds` to align with other `der_` families. 

@bundfussr I'm not entirely sure why the sections are not alphabetizing.  I posted onto RStudio Community forum about it.  Hoping someone can take a look.  I'm sure its a simple toggle but not apparent to me.

https://bms63.github.io/admiral/articles/programming_strategy.html#categorization-of-functions
- I moved the categorization of functions down to the end of the function section
- Tighten up some of the phrasing 

Could we make a go at finalizing this.  I'd like to move on from documentation!!

**Comment by bundfussr** [comment]
I guess the ordering issue is caused by
```
- subtitle: All ADaMs
- contents:
  - has_keyword("der_date_time")
  - has_keyword("der_gen")
```
Apparently the items listed in `contents:` are displayed in the specified order and ordered alphabetically only within an item.

So using just `der_gen` for all functions in "All ADaMs" should solve the issue.

By the way, why do we have two different keywords for the same section?

**Comment by bms63** [comment]
> By the way, why do we have two different keywords for the same section?

I was thinking that `der_date_time` functions could be used across different ADaMs.  However, I wanted to keep them separate so when you go look at a specific date/time function documentation you can see the other options for date/time.  Thinking that users might be interested in seeing the options for just date/time and not general adam derivations.

image

**Comment by bundfussr** [comment]
@bms63 , then we could use 
```
#' @family der_date_time
#'
#' @keywords der_gen der_date_time
```
in the header of the datetime derivations and in `_pkgdown.yml`
```
- subtitle: All ADaMs
- contents:
  - has_keyword("der_gen")
```

**Comment by bms63** [comment]
Living the dream!  253 files changed!!

**Comment by bms63** [comment]
@bundfussr @rossfarrugia @thomas-neitmann @morrishj  @koegerr do you all have time to review so we can get this merged in? Sounds like everyone is going to disappear again!!

**Comment by rossfarrugia** [comment]
Thanks @bms63! I already gave comments earlier that you incorporated so will bow out of this review and leave it to the others.

**Review by bundfussr — APPROVED** [review]

File: vignettes/programming_strategy.Rmd
Comment (bms63) [reply vignettes/programming_strategy.Rmd:411]: I think there is a `has_concept()` function that could do something similar to `has_keyword()`.  However, @bundfussr convinced to use two keywords for some of the more general functions, which helps with alphabetizing.  So I think we missed the boat on just having one or the other.

I'm going to change the note to:

 **NOTE:** It is strongly encouraged that each `@keyword` and `@family` are to be identical to ease burden of development and maintenance of `admiral` functions. If you need to use multiple keywords or families, please reach out to the core development team for discussion.

> @bms63 , then we could use
> 
> ```
> #' @family der_date_time
> #'
> #' @keywords der_gen der_date_time
> ```
> 
> in the header of the datetime derivations and in `_pkgdown.yml`
> 
> ```
> - subtitle: All ADaMs
> - contents:
>   - has_keyword("der_gen")
> ```

**Review by thomas-neitmann — APPROVED** [review]

**Comment by bms63** [comment]
@thomas-neitmann @rossfarrugia do you guys mind merging this into devel.  Github has it greyed out for me.  :( 

image

**Review by rossfarrugia — APPROVED** [review]
Approving to set off the check templates workflow

**Comment by rossfarrugia** [comment]
for me too @bms63 - you have the same admin rights as us now - i approved so that the check templates would run and then i should be able to merge this after that

**Pull request merged**
